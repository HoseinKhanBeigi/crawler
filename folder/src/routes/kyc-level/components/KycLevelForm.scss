.elementsContainer {
  display: flex;
  flex-wrap: wrap;
  gap: 8px;

  :global {
    .ant-tag-checkable {
      background-color: rgb(249, 249, 249);
      color: rgb(110, 110, 110);
      border: 1px solid rgb(217, 217, 217);
      margin: 0;
    }

    .ant-tag-checkable-checked {
      background-color: #1890ff;
      color: white;
      border: 1px solid transparent;
    }

    & .ant-tag-checkable:first-child {
      margin: 0 !important;
    }
  }
}

.buttonsContainer {
  display: flex;
  justify-content: flex-end;
  gap: 8px;
}
