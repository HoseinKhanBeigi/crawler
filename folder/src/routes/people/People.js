import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getPersonByMobileAction } from '../../store/person/person.actions';
import { getProductGroupsAction } from '../../store/getProducts/getProducts.actions';
import { columns, searchData } from './tableData';
import KianTable from '../../components/KianTable';
import withModal from '../../components/HOC/withModal';
import { PEOPLE_TABLE } from '../../store/settings/settings.constants';
import { kianTableApi } from '../../components/KianTable/helpers/globalApi';
import {
  DRAWER_FOR_GROUP_ASSIGN_TO,
  DRAWER_FOR_GROUP_TAGGING,
  DRAWER_FOR_SEND_GROUP_EMAIL,
  DRAWER_FOR_SEND_GROUP_MESSAGE,
  MODAL_FOR_DOWNLOAD_TABLE_EXCEL,
} from '../../components/ModalRoot/repository';
import { Actions } from '../../utils/aclActions';
import { BASE_VARIABLE_KEYS, resolveVariable } from '../../serviceConfig';
import history from '../../history';

const People = ({ productGroups, ...props }) => {
  const [selectedRowKeys, setSelectedRowKeys] = useState([]);
  const [selectedPeople, setSelectedPeople] = useState([]);
  const searchParam = history?.location?.search;

  useEffect(() => {
    props.getProductGroupsAction();
  }, []);

  const showModal = type => modalProps => () => {
    props.showModalAction({
      type,
      props: modalProps,
    });
  };

  const deSelectRows = () => {
    setSelectedRowKeys([]);
    setSelectedPeople([]);
  };

  const deselectAndRefreshTable = () => {
    deSelectRows();
    kianTableApi(PEOPLE_TABLE).refreshTable();
  };

  const rowSelection = {
    selectedRowKeys,
    onChange: (rows, selected) => {
      setSelectedPeople(selected.map(i => i.levantId));
      setSelectedRowKeys(rows);
    },
  };

  const contextMenu = [
    {
      label: 'ارسال پیام',
      icon: 'message',
      authority: Actions.sendSms,
      action: row =>
        showModal(DRAWER_FOR_SEND_GROUP_MESSAGE)({
          data: [row.levantId],
          deSelectRows,
        })(),
    },
    {
      label: 'ارسال ایمیل',
      icon: 'mail',
      authority: Actions.sendEmail,
      action: row =>
        showModal(DRAWER_FOR_SEND_GROUP_EMAIL)({
          selectedLevantIds: [row.levantId],
          deSelectRows,
        })(),
    },
    {
      label: 'برچسب زدن',
      icon: 'tag',
      authority: Actions.tagging,
      action: row =>
        showModal(DRAWER_FOR_GROUP_TAGGING)({
          taggedIds: [row.levantId],
          taggingClass: 'PEOPLE',
          name: `${row.firstName} ${row.lastName}`,
          deSelectRows: deselectAndRefreshTable,
        })(),
    },
    {
      label: 'اختصاص دادن',
      icon: 'check-circle',
      authority: Actions.leadAssign,
      action: row =>
        showModal(DRAWER_FOR_GROUP_ASSIGN_TO)({
          selectedLeads: [row.levantId],
          deSelectRows: deselectAndRefreshTable,
        })(),
    },
  ];

  const actionButtons = [
    {
      tooltip: `ارسال پیام به ${selectedPeople.length} فرد انتخابی`,
      icon: 'message',
      authority: Actions.sendSms,
      action: showModal(DRAWER_FOR_SEND_GROUP_MESSAGE)({
        data: selectedPeople,
        deSelectRows,
      }),
    },
    {
      tooltip: `ارسال ایمیل به ${selectedPeople.length} فرد انتخابی`,
      icon: 'mail',
      authority: Actions.sendEmail,
      action: showModal(DRAWER_FOR_SEND_GROUP_EMAIL)({
        selectedLevantIds: selectedPeople,
        deSelectRows,
      }),
    },
    {
      tooltip: `اختصاص دادن ${selectedPeople.length} فرد انتخابی`,
      icon: 'check-circle',
      authority: Actions.leadAssign,
      action: showModal(DRAWER_FOR_GROUP_ASSIGN_TO)({
        selectedLeads: selectedPeople,
        deSelectRows: deselectAndRefreshTable,
      }),
    },
    {
      tooltip: `برچسب زدن به ${selectedPeople.length} فرد انتخابی`,
      icon: 'tag',
      authority: Actions.tagging,
      action: showModal(DRAWER_FOR_GROUP_TAGGING)({
        taggedIds: selectedPeople,
        taggingClass: 'PEOPLE',
        deSelectRows: deselectAndRefreshTable,
      }),
    },
  ];

  const downloadExcelAction = () => () => {
    showModal(MODAL_FOR_DOWNLOAD_TABLE_EXCEL)({
      endpoint: `party/excel-report/person`,
      fileName: 'peopleReport',
      searchQuery: searchParam,
    });
  };

  return (
    <>
      <KianTable
        endpoint="party/search/person"
        searchData={searchData(productGroups)}
        showActionButtons={!!selectedPeople.length}
        actionButtons={actionButtons}
        rowSelection={rowSelection}
        contextMenu={contextMenu}
        downloadExcelAction={downloadExcelAction()}
        tableId={PEOPLE_TABLE}
        filterType="PERSON"
        headerTitle="افراد"
        columns={columns(resolveVariable(BASE_VARIABLE_KEYS.CONTEXT))}
        withSort={false}
      />
    </>
  );
};

People.propTypes = {
  getPersonByMobileAction: PropTypes.func.isRequired,
  getProductGroupsAction: PropTypes.func.isRequired,
  showModalAction: PropTypes.func.isRequired,
  products: PropTypes.array.isRequired,
  productGroups: PropTypes.array.isRequired,
};
const mapDispatch = {
  getPersonByMobileAction,
  getProductGroupsAction,
};

const mapState = ({ getProducts }) => ({
  productGroups: getProducts.productGroupsData,
});

export default connect(mapState, mapDispatch)(withModal(People));
export const peopleTest = People;
