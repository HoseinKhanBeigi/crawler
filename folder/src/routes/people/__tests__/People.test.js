/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import jest from 'jest-mock';
import Adapter from 'enzyme-adapter-react-16';
import merge from 'lodash/merge';
import ProviderMock from '../../../../tools/mocks/ProviderMock';
import { peopleTest as Component } from '../People';
import { AddPersonFormTest } from '../../../components/AddPersonForm';

configure({ adapter: new Adapter() });

describe('People Component', () => {
  const defaultProps = {
    getPersonAction: jest.fn(),
    getPersonByMobileAction: jest.fn().mockReturnValue('data'),
    getPersonInfoAction: jest.fn(),
    getActivitiesAction: jest.fn(),
    getOpportunityAction: jest.fn(),
    getTaskListAction: jest.fn(),
    personListError: false,
    personInfo: {
      id: 100,
    },
    activities: [],
    personList: {
      number: 0,
      totalElements: 1,
      size: 1,
      content: [
        {
          firstName: 'first_name',
          lastName: 'last_name',
          id: 10,
          levantId: 14,
          nationalCode: '123123',
          mobilePhone: '091234567890',
          key: 10,
        },
      ],
    },
  };
  /**
   * in this helper method, we provide the elements, which are often used
   * @returns {{editButton: *, column: (function(*=): *), row: *}}
   * @param propOverrides
   * @param isMount
   */
  const setup = (propOverrides = {}, isMount = false) => {
    const compNode = <Component {...defaultProps} {...propOverrides} />;
    const wrapper = isMount ? mount(ProviderMock(compNode)) : shallow(compNode);

    const row = wrapper.find('.ant-table-row-level-0').first();
    const column = index => row.find('td').at(index);

    const editButton = row
      .find('td')
      .last()
      .find('button');

    return { wrapper, row, editButton, column };
  };

  it('should render People component with the correct DOM structure', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render table rows correctly with all necessary fields', () => {
    const { row, column } = setup({}, 'mount');

    expect(row.prop('data-row-key')).toEqual(10);
    expect(row.find('td')).toHaveLength(6);
    expect(column(0).text()).toEqual('14');
    expect(column(1).text()).toEqual('first_name');
    expect(column(2).text()).toEqual('last_name');
    expect(column(3).text()).toEqual('091234567890');
    expect(column(4).text()).toEqual('123123');
  });

  it('should render error text, in case of any error', () => {
    const { wrapper } = setup({ personListError: true }, true);

    expect(wrapper.find('.alert')).toHaveLength(1);
    expect(wrapper.find('[type="error"]')).toHaveLength(2);
  });

  it('edit modal should open when correct states are there', () => {
    const { wrapper } = setup({}, true);
    const innerWrapper = wrapper.find('People').instance();

    expect(wrapper.find(AddPersonFormTest)).toHaveLength(0);

    innerWrapper.setState({ editPersonVisible: true });
    wrapper.update();
    expect(wrapper.find(AddPersonFormTest)).toHaveLength(1);

    innerWrapper.setState({
      editPersonVisible: false,
      addPersonVisible: false,
    });
    wrapper.update();
    expect(wrapper.find(AddPersonFormTest)).toHaveLength(0);

    innerWrapper.setState({ addPersonVisible: true });
    wrapper.update();
    expect(wrapper.find(AddPersonFormTest)).toHaveLength(1);
  });

  it('should show error instead of drawer if levantId is missing', done => {
    const newProps = merge(defaultProps, {
      personList: { content: [{ levantId: null }] },
    });
    const { row, wrapper } = setup(newProps, true);
    expect(row).toHaveLength(1);
    expect(wrapper.find('People').instance().state.visible).toBeFalsy();
    row.simulate('click');
    wrapper.update();
    // as the function includes, calling async props, if we do not use setTimeout, the state values will not change
    setTimeout(() => {
      expect(wrapper.find('People').instance().state.visible).toBeFalsy();
      done();
    }, 4000);
  });
});
