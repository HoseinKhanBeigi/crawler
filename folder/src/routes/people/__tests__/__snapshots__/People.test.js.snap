// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People Component should render People component with the correct DOM structure 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <People
    activities={Array []}
    getActivitiesAction={[MockFunction]}
    getOpportunityAction={[MockFunction]}
    getPersonAction={[MockFunction]}
    getPersonByMobileAction={[MockFunction]}
    getPersonInfoAction={[MockFunction]}
    getTaskListAction={[MockFunction]}
    personInfo={
      Object {
        "id": 100,
      }
    }
    personList={
      Object {
        "content": Array [
          Object {
            "firstName": "first_name",
            "id": 10,
            "key": 10,
            "lastName": "last_name",
            "levantId": 14,
            "mobilePhone": "091234567890",
            "nationalCode": "123123",
          },
        ],
        "number": 0,
        "size": 1,
        "totalElements": 1,
      }
    }
    personListError={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Toolbar
          addNewDropDown={true}
          addNewDropDownFunc={[Function]}
          addNewDropDownOverlay={
            Array [
              Object {
                "key": 1,
                "name": "افزودن فرد",
                "value": "plus",
              },
              Object {
                "key": 2,
                "name": "وارد کردن افراد",
                "value": "import",
              },
            ]
          }
          allDropDownFunc={[Function]}
          allDropDownOverlay={null}
          allDropDownTitle="تمام نخ ها"
          contactFunc={[Function]}
          contactOverlay={null}
          drawerChildren={
            <b>
              محتوای تستی
            </b>
          }
          drawerTitle="فیلتر نخ ها"
          hasFilter={false}
          hasSearch={false}
          settingFunc={[Function]}
          settingOverlay={null}
          title={null}
        />,
        <AdvancedSearchBar
          className={null}
          data={
            Array [
              Object {
                "name": "levantId",
                "title": "شناسه لوانت",
                "type": "input",
              },
              Object {
                "name": "firstName",
                "title": "نام",
                "type": "input",
              },
              Object {
                "name": "lastName",
                "title": "نام خانوادگی",
                "type": "input",
              },
              Object {
                "name": "mobilePhone",
                "title": "شماره همراه",
                "type": "input",
              },
              Object {
                "name": "nationalCode",
                "title": "کد ملی",
                "type": "input",
              },
            ]
          }
          onSearchLead={[Function]}
        />,
        <CPTable
          bodyStyle={undefined}
          bordered={true}
          childrenColumnName={undefined}
          className={undefined}
          columns={
            Array [
              Object {
                "dataIndex": "levantId",
                "key": "levantId",
                "render": [Function],
                "title": "شناسه لوانت",
              },
              Object {
                "dataIndex": "firstName",
                "key": "firstName",
                "title": "نام",
              },
              Object {
                "dataIndex": "lastName",
                "key": "lastName",
                "title": "نام خانوادگی",
              },
              Object {
                "dataIndex": "mobilePhone",
                "key": "mobilePhone",
                "title": "شماره همراه",
              },
              Object {
                "dataIndex": "nationalCode",
                "key": "nationalCode",
                "title": "کد ملی",
              },
              Object {
                "dataIndex": "mobilePhone",
                "key": "actions",
                "render": [Function],
                "title": "عملیات",
                "width": 150,
              },
            ]
          }
          components={undefined}
          data={
            Array [
              Object {
                "firstName": "first_name",
                "id": 10,
                "key": 10,
                "lastName": "last_name",
                "levantId": 14,
                "mobilePhone": "091234567890",
                "nationalCode": "123123",
              },
            ]
          }
          defaultExpandAllRows={false}
          defaultExpandedRowKeys={undefined}
          expandIconAsCell={true}
          expandIconColumnIndex={undefined}
          expandRowByClick={true}
          expandedRowKeys={Array []}
          expandedRowRender={null}
          footer={null}
          indentSize={undefined}
          loading={false}
          locale={undefined}
          onChange={[Function]}
          onExpand={[Function]}
          onExpandedRowsChange={[Function]}
          onHeaderRow={[Function]}
          onRow={[Function]}
          onRowClick={undefined}
          pagination={false}
          rowClassName={undefined}
          rowKey="id"
          rowSelection={null}
          scroll={undefined}
          showHeader={true}
          size="default"
          style={undefined}
          title={null}
          useFixedHeader={false}
        />,
        <div
          className="pagination"
        >
          <CPPanel
            current={1}
            onChange={[Function]}
            onShowSizeChange={[Function]}
            pageSize={1}
            showQuickJumper={true}
            showSizeChanger={true}
            size="small"
            total={1}
          />
        </div>,
        <Connect(EntityDrawer)
          handleVisible={[Function]}
          source="PERSON"
          visible={false}
        />,
        <CPModal
          afterClose={undefined}
          className="modalAddPerson"
          closable={true}
          confirmLoading={false}
          footer={null}
          mask={true}
          maskClosable={true}
          onCancel={[Function]}
          onOk={[Function]}
          title="ویرایش کاربر"
          visible={false}
          width={520}
        />,
        <CPModal
          afterClose={undefined}
          className="min_modal"
          closable={true}
          confirmLoading={false}
          footer={null}
          mask={true}
          maskClosable={true}
          onCancel={[Function]}
          onOk={[Function]}
          title=""
          visible={false}
          width={520}
        >
          <span>
            این امکان در حال حاضر وجود ندارد ...
          </span>
        </CPModal>,
      ],
      "className": "people",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "addNewDropDown": true,
          "addNewDropDownFunc": [Function],
          "addNewDropDownOverlay": Array [
            Object {
              "key": 1,
              "name": "افزودن فرد",
              "value": "plus",
            },
            Object {
              "key": 2,
              "name": "وارد کردن افراد",
              "value": "import",
            },
          ],
          "allDropDownFunc": [Function],
          "allDropDownOverlay": null,
          "allDropDownTitle": "تمام نخ ها",
          "contactFunc": [Function],
          "contactOverlay": null,
          "drawerChildren": <b>
            محتوای تستی
          </b>,
          "drawerTitle": "فیلتر نخ ها",
          "hasFilter": false,
          "hasSearch": false,
          "settingFunc": [Function],
          "settingOverlay": null,
          "title": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": null,
          "data": Array [
            Object {
              "name": "levantId",
              "title": "شناسه لوانت",
              "type": "input",
            },
            Object {
              "name": "firstName",
              "title": "نام",
              "type": "input",
            },
            Object {
              "name": "lastName",
              "title": "نام خانوادگی",
              "type": "input",
            },
            Object {
              "name": "mobilePhone",
              "title": "شماره همراه",
              "type": "input",
            },
            Object {
              "name": "nationalCode",
              "title": "کد ملی",
              "type": "input",
            },
          ],
          "onSearchLead": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bodyStyle": undefined,
          "bordered": true,
          "children": undefined,
          "childrenColumnName": undefined,
          "className": undefined,
          "columns": Array [
            Object {
              "dataIndex": "levantId",
              "key": "levantId",
              "render": [Function],
              "title": "شناسه لوانت",
            },
            Object {
              "dataIndex": "firstName",
              "key": "firstName",
              "title": "نام",
            },
            Object {
              "dataIndex": "lastName",
              "key": "lastName",
              "title": "نام خانوادگی",
            },
            Object {
              "dataIndex": "mobilePhone",
              "key": "mobilePhone",
              "title": "شماره همراه",
            },
            Object {
              "dataIndex": "nationalCode",
              "key": "nationalCode",
              "title": "کد ملی",
            },
            Object {
              "dataIndex": "mobilePhone",
              "key": "actions",
              "render": [Function],
              "title": "عملیات",
              "width": 150,
            },
          ],
          "components": undefined,
          "data": Array [
            Object {
              "firstName": "first_name",
              "id": 10,
              "key": 10,
              "lastName": "last_name",
              "levantId": 14,
              "mobilePhone": "091234567890",
              "nationalCode": "123123",
            },
          ],
          "defaultExpandAllRows": false,
          "defaultExpandedRowKeys": undefined,
          "expandIconAsCell": true,
          "expandIconColumnIndex": undefined,
          "expandRowByClick": true,
          "expandedRowKeys": Array [],
          "expandedRowRender": null,
          "footer": null,
          "indentSize": undefined,
          "loading": false,
          "locale": undefined,
          "onChange": [Function],
          "onExpand": [Function],
          "onExpandedRowsChange": [Function],
          "onHeaderRow": [Function],
          "onRow": [Function],
          "onRowClick": undefined,
          "pagination": false,
          "rowClassName": undefined,
          "rowKey": "id",
          "rowSelection": null,
          "scroll": undefined,
          "showHeader": true,
          "size": "default",
          "style": undefined,
          "title": null,
          "useFixedHeader": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CPPanel
            current={1}
            onChange={[Function]}
            onShowSizeChange={[Function]}
            pageSize={1}
            showQuickJumper={true}
            showSizeChanger={true}
            size="small"
            total={1}
          />,
          "className": "pagination",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "current": 1,
            "onChange": [Function],
            "onShowSizeChange": [Function],
            "pageSize": 1,
            "showQuickJumper": true,
            "showSizeChanger": true,
            "size": "small",
            "total": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleVisible": [Function],
          "source": "PERSON",
          "visible": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "afterClose": undefined,
          "children": false,
          "className": "modalAddPerson",
          "closable": true,
          "confirmLoading": false,
          "footer": null,
          "mask": true,
          "maskClosable": true,
          "onCancel": [Function],
          "onOk": [Function],
          "title": "ویرایش کاربر",
          "visible": false,
          "width": 520,
        },
        "ref": null,
        "rendered": false,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "afterClose": undefined,
          "children": <span>
            این امکان در حال حاضر وجود ندارد ...
          </span>,
          "className": "min_modal",
          "closable": true,
          "confirmLoading": false,
          "footer": null,
          "mask": true,
          "maskClosable": true,
          "onCancel": [Function],
          "onOk": [Function],
          "title": "",
          "visible": false,
          "width": 520,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "این امکان در حال حاضر وجود ندارد ...",
          },
          "ref": null,
          "rendered": "این امکان در حال حاضر وجود ندارد ...",
          "type": "span",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Toolbar
            addNewDropDown={true}
            addNewDropDownFunc={[Function]}
            addNewDropDownOverlay={
              Array [
                Object {
                  "key": 1,
                  "name": "افزودن فرد",
                  "value": "plus",
                },
                Object {
                  "key": 2,
                  "name": "وارد کردن افراد",
                  "value": "import",
                },
              ]
            }
            allDropDownFunc={[Function]}
            allDropDownOverlay={null}
            allDropDownTitle="تمام نخ ها"
            contactFunc={[Function]}
            contactOverlay={null}
            drawerChildren={
              <b>
                محتوای تستی
              </b>
            }
            drawerTitle="فیلتر نخ ها"
            hasFilter={false}
            hasSearch={false}
            settingFunc={[Function]}
            settingOverlay={null}
            title={null}
          />,
          <AdvancedSearchBar
            className={null}
            data={
              Array [
                Object {
                  "name": "levantId",
                  "title": "شناسه لوانت",
                  "type": "input",
                },
                Object {
                  "name": "firstName",
                  "title": "نام",
                  "type": "input",
                },
                Object {
                  "name": "lastName",
                  "title": "نام خانوادگی",
                  "type": "input",
                },
                Object {
                  "name": "mobilePhone",
                  "title": "شماره همراه",
                  "type": "input",
                },
                Object {
                  "name": "nationalCode",
                  "title": "کد ملی",
                  "type": "input",
                },
              ]
            }
            onSearchLead={[Function]}
          />,
          <CPTable
            bodyStyle={undefined}
            bordered={true}
            childrenColumnName={undefined}
            className={undefined}
            columns={
              Array [
                Object {
                  "dataIndex": "levantId",
                  "key": "levantId",
                  "render": [Function],
                  "title": "شناسه لوانت",
                },
                Object {
                  "dataIndex": "firstName",
                  "key": "firstName",
                  "title": "نام",
                },
                Object {
                  "dataIndex": "lastName",
                  "key": "lastName",
                  "title": "نام خانوادگی",
                },
                Object {
                  "dataIndex": "mobilePhone",
                  "key": "mobilePhone",
                  "title": "شماره همراه",
                },
                Object {
                  "dataIndex": "nationalCode",
                  "key": "nationalCode",
                  "title": "کد ملی",
                },
                Object {
                  "dataIndex": "mobilePhone",
                  "key": "actions",
                  "render": [Function],
                  "title": "عملیات",
                  "width": 150,
                },
              ]
            }
            components={undefined}
            data={
              Array [
                Object {
                  "firstName": "first_name",
                  "id": 10,
                  "key": 10,
                  "lastName": "last_name",
                  "levantId": 14,
                  "mobilePhone": "091234567890",
                  "nationalCode": "123123",
                },
              ]
            }
            defaultExpandAllRows={false}
            defaultExpandedRowKeys={undefined}
            expandIconAsCell={true}
            expandIconColumnIndex={undefined}
            expandRowByClick={true}
            expandedRowKeys={Array []}
            expandedRowRender={null}
            footer={null}
            indentSize={undefined}
            loading={false}
            locale={undefined}
            onChange={[Function]}
            onExpand={[Function]}
            onExpandedRowsChange={[Function]}
            onHeaderRow={[Function]}
            onRow={[Function]}
            onRowClick={undefined}
            pagination={false}
            rowClassName={undefined}
            rowKey="id"
            rowSelection={null}
            scroll={undefined}
            showHeader={true}
            size="default"
            style={undefined}
            title={null}
            useFixedHeader={false}
          />,
          <div
            className="pagination"
          >
            <CPPanel
              current={1}
              onChange={[Function]}
              onShowSizeChange={[Function]}
              pageSize={1}
              showQuickJumper={true}
              showSizeChanger={true}
              size="small"
              total={1}
            />
          </div>,
          <Connect(EntityDrawer)
            handleVisible={[Function]}
            source="PERSON"
            visible={false}
          />,
          <CPModal
            afterClose={undefined}
            className="modalAddPerson"
            closable={true}
            confirmLoading={false}
            footer={null}
            mask={true}
            maskClosable={true}
            onCancel={[Function]}
            onOk={[Function]}
            title="ویرایش کاربر"
            visible={false}
            width={520}
          />,
          <CPModal
            afterClose={undefined}
            className="min_modal"
            closable={true}
            confirmLoading={false}
            footer={null}
            mask={true}
            maskClosable={true}
            onCancel={[Function]}
            onOk={[Function]}
            title=""
            visible={false}
            width={520}
          >
            <span>
              این امکان در حال حاضر وجود ندارد ...
            </span>
          </CPModal>,
        ],
        "className": "people",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addNewDropDown": true,
            "addNewDropDownFunc": [Function],
            "addNewDropDownOverlay": Array [
              Object {
                "key": 1,
                "name": "افزودن فرد",
                "value": "plus",
              },
              Object {
                "key": 2,
                "name": "وارد کردن افراد",
                "value": "import",
              },
            ],
            "allDropDownFunc": [Function],
            "allDropDownOverlay": null,
            "allDropDownTitle": "تمام نخ ها",
            "contactFunc": [Function],
            "contactOverlay": null,
            "drawerChildren": <b>
              محتوای تستی
            </b>,
            "drawerTitle": "فیلتر نخ ها",
            "hasFilter": false,
            "hasSearch": false,
            "settingFunc": [Function],
            "settingOverlay": null,
            "title": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": null,
            "data": Array [
              Object {
                "name": "levantId",
                "title": "شناسه لوانت",
                "type": "input",
              },
              Object {
                "name": "firstName",
                "title": "نام",
                "type": "input",
              },
              Object {
                "name": "lastName",
                "title": "نام خانوادگی",
                "type": "input",
              },
              Object {
                "name": "mobilePhone",
                "title": "شماره همراه",
                "type": "input",
              },
              Object {
                "name": "nationalCode",
                "title": "کد ملی",
                "type": "input",
              },
            ],
            "onSearchLead": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bodyStyle": undefined,
            "bordered": true,
            "children": undefined,
            "childrenColumnName": undefined,
            "className": undefined,
            "columns": Array [
              Object {
                "dataIndex": "levantId",
                "key": "levantId",
                "render": [Function],
                "title": "شناسه لوانت",
              },
              Object {
                "dataIndex": "firstName",
                "key": "firstName",
                "title": "نام",
              },
              Object {
                "dataIndex": "lastName",
                "key": "lastName",
                "title": "نام خانوادگی",
              },
              Object {
                "dataIndex": "mobilePhone",
                "key": "mobilePhone",
                "title": "شماره همراه",
              },
              Object {
                "dataIndex": "nationalCode",
                "key": "nationalCode",
                "title": "کد ملی",
              },
              Object {
                "dataIndex": "mobilePhone",
                "key": "actions",
                "render": [Function],
                "title": "عملیات",
                "width": 150,
              },
            ],
            "components": undefined,
            "data": Array [
              Object {
                "firstName": "first_name",
                "id": 10,
                "key": 10,
                "lastName": "last_name",
                "levantId": 14,
                "mobilePhone": "091234567890",
                "nationalCode": "123123",
              },
            ],
            "defaultExpandAllRows": false,
            "defaultExpandedRowKeys": undefined,
            "expandIconAsCell": true,
            "expandIconColumnIndex": undefined,
            "expandRowByClick": true,
            "expandedRowKeys": Array [],
            "expandedRowRender": null,
            "footer": null,
            "indentSize": undefined,
            "loading": false,
            "locale": undefined,
            "onChange": [Function],
            "onExpand": [Function],
            "onExpandedRowsChange": [Function],
            "onHeaderRow": [Function],
            "onRow": [Function],
            "onRowClick": undefined,
            "pagination": false,
            "rowClassName": undefined,
            "rowKey": "id",
            "rowSelection": null,
            "scroll": undefined,
            "showHeader": true,
            "size": "default",
            "style": undefined,
            "title": null,
            "useFixedHeader": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CPPanel
              current={1}
              onChange={[Function]}
              onShowSizeChange={[Function]}
              pageSize={1}
              showQuickJumper={true}
              showSizeChanger={true}
              size="small"
              total={1}
            />,
            "className": "pagination",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "current": 1,
              "onChange": [Function],
              "onShowSizeChange": [Function],
              "pageSize": 1,
              "showQuickJumper": true,
              "showSizeChanger": true,
              "size": "small",
              "total": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleVisible": [Function],
            "source": "PERSON",
            "visible": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "afterClose": undefined,
            "children": false,
            "className": "modalAddPerson",
            "closable": true,
            "confirmLoading": false,
            "footer": null,
            "mask": true,
            "maskClosable": true,
            "onCancel": [Function],
            "onOk": [Function],
            "title": "ویرایش کاربر",
            "visible": false,
            "width": 520,
          },
          "ref": null,
          "rendered": false,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "afterClose": undefined,
            "children": <span>
              این امکان در حال حاضر وجود ندارد ...
            </span>,
            "className": "min_modal",
            "closable": true,
            "confirmLoading": false,
            "footer": null,
            "mask": true,
            "maskClosable": true,
            "onCancel": [Function],
            "onOk": [Function],
            "title": "",
            "visible": false,
            "width": 520,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "این امکان در حال حاضر وجود ندارد ...",
            },
            "ref": null,
            "rendered": "این امکان در حال حاضر وجود ندارد ...",
            "type": "span",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
